src/Mod/Sketcher/Gui/CommandConstraints.cpp

```diff
@@ -2725,443 +2721,304 @@ void CmdSketcherConstrainHorizontal::activated(int iMsg)
-                                  *it);
-        }
-    }
-    else if (fixedpoints <= 1) {// vertex mode, maximum one fixed point
-        // undo command open
-        openCommand(QT_TRANSLATE_NOOP("Command", "Add vertical alignment"));
-        std::vector<int>::iterator it;
-        std::vector<Sketcher::PointPos>::iterator itp;
-        for (it = pointgeoids.begin(), itp = pointpos.begin();
-             it != std::prev(pointgeoids.end()) && itp != std::prev(pointpos.end());
-             it++, itp++) {
-            // issue the actual commands to create the constraint
-            Gui::cmdAppObjectArgs(selection[0].getObject(),
-                                  "addConstraint(Sketcher.Constraint('Vertical',%d,%d,%d,%d))",
-                                  *it,
-                                  static_cast<int>(*itp),
-                                  *std::next(it),
-                                  static_cast<int>(*std::next(itp)));
-        }
-    }
-    else {// vertex mode, fixedpoints > 1
-        Gui::TranslatedUserWarning(Obj,
-                                   QObject::tr("Impossible constraint"),
-                                   QObject::tr("There are more than one fixed points selected. "
-                                               "Select a maximum of one fixed point!"));
-        return;
-    }
+protected:
+    void activated(int iMsg) override;
+    void applyConstraint(std::vector<SelIdPair>& selSeq, int seqIndex) override;
+};
 
-    // finish the transaction and update
-    commitCommand();
+CmdSketcherConstrainVertical::CmdSketcherConstrainVertical()
+    : CmdSketcherConstraint("Sketcher_ConstrainVertical")
+{
+    sAppModule = "Sketcher";
+    sGroup = "Sketcher";
+    sMenuText = QT_TR_NOOP("Constrain vertically");
+    sToolTipText = QT_TR_NOOP("Create a vertical constraint on the selected item");
+    sWhatsThis = "Sketcher_ConstrainVertical";
+    sStatusTip = sToolTipText;
+    sPixmap = "Constraint_Vertical";
+    sAccel = "V";
+    eType = ForEdit;
 
-    tryAutoRecompute(Obj);
+    allowedSelSequences = {{SelEdge}, {SelVertex, SelVertexOrRoot}, {SelRoot, SelVertex}};
+}
 
-    // clear the selection (convenience)
-    getSelection().clearSelection();
+void CmdSketcherConstrainVertical::activated(int iMsg)
+{
+    Q_UNUSED(iMsg);
+    horVerActivated(this, "Vertical");
 }
 
 void CmdSketcherConstrainVertical::applyConstraint(std::vector<SelIdPair>& selSeq, int seqIndex)
 {
-    SketcherGui::ViewProviderSketch* sketchgui =
-        static_cast<SketcherGui::ViewProviderSketch*>(getActiveGuiDocument()->getInEdit());
-    Sketcher::SketchObject* Obj = sketchgui->getSketchObject();
-
-    switch (seqIndex) {
-        case 0:// {Edge}
-        {
-            // create the constraint
-            const std::vector<Sketcher::Constraint*>& vals = Obj->Constraints.getValues();
-
-            int CrvId = selSeq.front().GeoId;
```

Comment:
@wwmayer regarding your comment on the other PR, this line of code was already like this before. So the auto vertical/horizontal PR did not changed that.
And it seemed that it wasn't causing problem before. So I'm not sure if it problematic? What do you think?